static/js/import_wizard/utils/detection.ts

```diff
@@ -0,0 +1,204 @@
+/**
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import _ from "lodash";
+
+import {
+  Column,
+  DCProperty,
+  DCType,
+  MappedThing,
+  Mapping,
+  MappingType,
+  MappingVal,
+  TypeProperty,
+} from "../types";
+
+const COLUMN_KEY = "column";
+const COLUMN_IDX_KEY = "columnIdx";
+const DISP_NAME_KEY = "displayName";
+const PLACETYPE_KEY = "placeType";
+const PLACEPROP_KEY = "placeProperty";
+const DCTYPE_KEY = "dcType";
+const DCPROP_KEY = "dcProperty";
+const HEADER_KEY = "headers";
+
+const COLUMN_EMPTY: Column = { id: "", header: "", columnIdx: 0 };
+const DCPROPERTY_EMPTY: DCProperty = { dcid: "", displayName: "" };
+const DCTYPE_EMPTY: DCType = { dcid: "", displayName: "" };
+
+function _replaceStrings(originalString: string): string {
+  // Need to replace specific strings. Cannot do a blanket regex replacement
+  // because it could end up replacing user provided values which we do not
+  // want to edit. This only replaces the keys/fields which are different in
+  // the API response compared to the attributes of the corresponding Types.
+  originalString = originalString.split("column_idx").join(COLUMN_IDX_KEY);
+  originalString = originalString.split("display_name").join(DISP_NAME_KEY);
+  originalString = originalString.split("place_type").join(PLACETYPE_KEY);
+  originalString = originalString.split("place_property").join(PLACEPROP_KEY);
+  originalString = originalString.split("dc_type").join(DCTYPE_KEY);
+  originalString = originalString.split("dc_property").join(DCPROP_KEY);
+  return originalString;
+}
+
+/**
+ * Given the detection json response from the detection server API,
+ * parse it in to a Mapping structure.
+ *
+ * @param detectedJSONString is the JSON response from the server API.
```

Comment:
the response from the server API will actually be an object. So what we want to do is:
1. create a new type like: https://github.com/datacommonsorg/website/blob/master/static/js/shared/stat_types.ts#L64
2. this function should actually be a function to convert from the API returned object to a Mapping object.

Same with the other parse function

Reply:
Removed the string parsing like discussed offline.